WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -604,7 +604,7 @@ def refund_payment
 
   def create
     @competition = competition_from_params
-    if !@competition.registration_opened?
+    if !@competition.registration_opened? && !@competition.user_can_pre_register?(current_user)
```

Comment:
Ruby nit: `if !foo` should become `unless foo`. To be fair, this was already wrong before your commit, but now that you're touching this line might as well clean it up :upside_down_face: 

`unless @competition.registration_opened? || @competition.user_can_pre_register?(current_user)`

Reply:
Nifty, I have made this change